gm4_metallurgy/data/gm4_vibro_shamir/functions/shock/calculate.mcfunction

```diff
@@ -0,0 +1,18 @@
+# @s = mob within 6 blocks of a vibro shockwave
+# run from shock/activate
+
+# raycast to the vibro player
+summon marker ~ ~ ~ {CustomName:'"gm4_vibro_ray"',Tags:["gm4_vibro_ray"]}
+execute as @e[type=marker,tag=gm4_vibro_ray,limit=1] at @s facing entity @a[tag=gm4_vibro_user,limit=1] feet run tp @s ~ ~ ~ ~ ~
+scoreboard players reset $ray gm4_vibro_shock
+execute as @e[type=marker,tag=gm4_vibro_ray,limit=1] at @s run function gm4_vibro_shamir:shock/ray
+kill @e[type=marker,tag=gm4_vibro_ray]
```

Comment:
You could cute the amount of `@e` calls by 66% if you moved these into a common function.

Reply:
This is how we've always done ray casting, but you're right; I'll update it